### DISCUSSION TITLE
microCOVID app concept

### DISCUSSION THREAD

**Discussion by xbruge** [microCOVID/microCOVID#1637] [Show and tell]
As a UX design student, I thought it might be nice to have a simplified mobile app for microCOVID where you could not only calculate risks on the go/offline, but also save them into a budgeting diary. It could also be helpful to save activities that you partake in frequently to save you from recalculating the risk. 

The Diary tab displays a rounded percentage for prevalence for simplicity, along with a calendar overview of the recent month. Green dots denote scenarios added that don't end up exceeding the weekly budget, whereas an orange dot means you went over. For the Calculator tab I decided to simplify the pickers to display only the most necessary info, all the explanations and research would be hidden but still accessible with the 'i' info button at the top. One of the hardest things about the pandemic that I only feel is getting worse as more people return to "normal" is friendship and social connections. Looking through my Instagram stories, I've not spotted a single mask since protections have been removed here. I've been trying to communicate the risks of long COVID to my friends but I just sound extremely dissonant to them, they've moved on. I guess in the hopes I find covid cautious friends someday, the Friends tab would be a way to keep motivated and compete with friends. The number in the green dots represent how many months the person has had a streak of staying within budget. The percentage beside the name is their yearly budget. You can add a "current status" or add your favourite masks under your name for a bit of personality/fun. 

I'd love to hear any thoughts/feedback, if we can find a developer (no clue how complex this would be to make) I think it would be a really cool tool, especially for those who perhaps might be new to calculating Covid risks and/or prefer a mobile/offline experience. 

Thanks to those that have contributed for such a useful site!
Bruce

![microCOVID app](https://user-images.githubusercontent.com/51087242/201022105-e45b0c6e-9839-4a29-9085-a152716ed66f.png)

**Comment by apiology** [comment]
I love this!

**Comment by apiology** [comment]
I think the process of persistently and accurately logging multiple exposures throughout the day is a rich vein of UX challenges.  No one likes data entry, and the faster, more automatic, and less error prone it can be made, the better.

The [risk tracker spreadsheet](https://www.microcovid.org/tracker) is one attempt with some clever approaches, but clearly a mobile app is going to be much more accessible in the moment.

I've used one of the [popular CO2 monitors](https://aranet.com/products/aranet4/) in my home.  The monitor has a protocol folks have [tied into](https://github.com/Anrijs/Aranet4-Python).  I've considered wearing that monitor and reviewing that graph periodically for any spikes to make sure I haven't missed logging microCOVIDs.  

In principle, we could do that a little more automatically and start draft log entries automatically--input from it it could be used as a proxy for a number of other variables and reduce the number of things the user would need to remember about a loggable scenario.   We'd need to do some research to get a defensible conversion factor, certainly.

<img src="https://user-images.githubusercontent.com/3681194/201474385-8b42abaf-8f2b-4fa6-8b97-77872f5b5159.png" alt="mobile app screenshot of co2 graph" width="200"/>

**Reply by xbruge** [reply]
Yeah, constantly logging various scenarios with different variables is definitely a headache.

Interesting idea to use CO2 data from an Aranet4 to log entries, reminds me of the 'Automatic Workout Detection' on the Apple Watch. Although I must admit I turned that feature off as I found it to be hit or miss. 

Would be cool if it was accurate enough so that one only had to enter 3 fields: 'People within 5m' (this could also have a picker with options such as 1/80/700 but with the option to enter a custom value), 'Distance' and 'Their volume' with the other fields such as masking making assumptions based on past behaviour (eg. most in my area are not masking, and I am always in a 3M Aura). I think that would make for a handy Watch app. 

Sadly, I can't help with any research or software dev but wish I had the skills to!

![auto workout detect](https://user-images.githubusercontent.com/51087242/201557726-3d70a657-d2e5-4e14-a402-3152b77c7f4a.png)

**Reply by apiology** [reply]
Great analogy!  I use the Automatic Workout Detection feature all the time when I forget to start the mode myself when I go out for a walk, and yeah, I think watch notifications could be a great way to trigger the user into logging.

I was also envisioning an end of day mode where the user might review the spikes in the graph and be able to easily add detail to those.  But the more spikes we can catch at the time they happen, the more accurate the data entry will be.

**Reply by apiology** [reply]
To give you an idea of what may or may not be possible, here's some data from me yesterday, along with my rough microCOVID measures of the same events I made before looking at the graph:

![image](https://user-images.githubusercontent.com/3681194/201935365-c4b6387b-8a83-432b-976f-554356a96193.png)

Roughly annotated:
* 3am-9am: My home office (empty) - (0)
* 9am-3:45pm: My home office (with me working at my desk) (0)
* 3:55pm-4:15pm - Uber ride to airport (yikes!) (2)
* 4:15pm-4:45pm - Chilling in a remote hallway at the Airport (1)
* 4:45pm-6:10pm - Going through security and then in a crowded terminal with high ceilings (37)
* 6:10pm-8:30pm On board plane - about 30 minutes on the ground before taking off, maybe 20 after landing (39)
* 8:30pm-9:00pm - Working way through airport to shared ride pickup (2)
* 9pm-9:20pm - Uber from airport to hotel (4)
* 9:20pm- Hotel room (0)

CSV: [Raw data exported from the app](https://gist.github.com/apiology/88ede442f722ba89ec6bc300606ef086)

**Reply by apiology** [reply]
Oh, it also looks like you can get phone GPS info from your friendly panopticon-keepers: https://timeline.google.com 

If it could be integrated in, GPS data could help prompt the user with place names associated with the spikes to jog their memory.  It could also eliminate false spikes when the user was at home with folks who are already accounted for in their household risk.

**Reply by lazerwalker** [reply]
I can't speak to Google Timeline, but I can say that Apple's place detection APIs are decent but imperfect. Manually applying clustering algorithms to raw GPS data to do place detection yourself is a major pain, so my gut would be to see if Google/Apple's built-in APIs are good enough.

I'm also not sure if Google or someone else offers a free API to translate GPS coordinates to venue names. Foursquare used to be the gold standard, but they've shifted away from providing that as a free (or reasonable to small-scale project) service.

**Reply by apiology** [reply]
Ah, nice! I only mention Timeline because it's something I'm using in my little prototype of this concept, where I do a daily check of both my Timeline map and my Aranet4 graph for the last day and reconstruct my COVID exposure using microCOVID, to make sure I'm keeping roughly in the budget I set myself.  That said, it's pretty good at identifying times when you've stopped at a location, and prompts you with suggestions on what that location actually was (e.g.,  post office, which store, etc).

**Comment by lazerwalker** [comment]
I have a decent chunk of domain expertise in building self-tracking apps like this (a lot of experience on native iOS, some amount using cross-platform web tech, a very small amount using native Android). I don't think I have the energy to take the lead on building something like this right now (although this may change), but if other developers come out of the woodwork I'm happy to make small contributions and/or provide mentorship if folks are less experienced with mobile app development specifically.

**Reply by apiology** [reply]
My ambitions far outpace my available free time, but who knows, I may take you up on that offer at some point 🤣
